generator/internal/dart/templates/lib/method.mustache

```diff
@@ -17,10 +17,31 @@ limitations under the License.
 {{#Codec.DocLines}}
 {{{.}}}
 {{/Codec.DocLines}}
-Future<{{Codec.ResponseType}}> {{Codec.Name}}({{Codec.RequestType}} request) async {
+{{#OperationInfo}}
+///
+/// Returns an [Operation] representing the status of the long-running
+/// operation.
+///
+/// When complete, [Operation.done] will be `true`. If successful,
+/// [Operation.responseAsMessage] will contain the Operation's result.
+{{/OperationInfo}}
+{{#Codec.IsLROGetOperation}}
+///
+/// This method can be used to get the current status of a long-running
+/// operation.
+Future<Operation<T, S>> getOperation<T extends Message, S extends Message>(Operation<T, S> request) async {
+{{/Codec.IsLROGetOperation}}
+{{^Codec.IsLROGetOperation}}
+Future<{{Codec.ResponseType}}{{#OperationInfo}}<{{Codec.ResponseType}}, {{Codec.MetadataType}}>{{/OperationInfo}}> {{Codec.Name}}({{Codec.RequestType}} request) async {
+{{/Codec.IsLROGetOperation}}
   final url = Uri.https(_host, '{{PathInfo.Codec.PathFmt}}');
   {{#Codec.ReturnsValue}}final response = {{/Codec.ReturnsValue}}await _client.{{Codec.RequestMethod}}(url{{#Codec.HasBody}}, body: {{Codec.BodyMessageName}}{{/Codec.HasBody}});
-{{#Codec.ReturnsValue}}
-  return {{Codec.ResponseType}}.fromJson(response);
-{{/Codec.ReturnsValue}}
+{{#Codec.IsLROGetOperation}}
```

Comment:
Ugh... I wonder if it would be cleared to just eat the 3 duplicated lines (or move them to a partial if we must) and have a single `{{#Codec.IsLROGetOperation}}` and a single `{{^Codec.IsLROGetoperation}}`?

Do you mind giving that a try? I trust you to make the right call once you look at the alternatives.

Reply:
Will do! I'm fine w/ a little bit of copied code if it makes the templates easier to read.